### PR TITLE
Errorlevels

### PR DESCRIPTION
**Pull request by wordshaker**
A post on the different error levels.
I can firm this up to what error levels mean to Oakbrook once I've had some feedback

### CODE REVIEW

File: _posts/2017-03-11-Error-Levels.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+#### FATAL
+
+THE WORLD IS ENDING! This is for the worst of the very worst situation. This is for situations which force the application to close.
+As such their use should be very rare, if used at all. 
+
+#### ERROR
+
+This is for something that has gone wrong that will not force the application to close but it will cause issues. These are for logging errors 
+taht could cause the operation to close, but can also be used for issues that don't cause an application to close (i.e logging a 500 from a request).
```
Comment (neonboxx) [inline _posts/2017-03-11-Error-Levels.md]: that

File: _posts/2017-03-11-Error-Levels.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+#### ERROR
+
+This is for something that has gone wrong that will not force the application to close but it will cause issues. These are for logging errors 
+taht could cause the operation to close, but can also be used for issues that don't cause an application to close (i.e logging a 500 from a request). 
+It is for any problems that require immediate intervention. 
+
+#### WARN
+
+This is for problems that do not require immediate intervention, but that may cause issues or peculiarities at some point. It provide information
```
Comment (neonboxx) [inline _posts/2017-03-11-Error-Levels.md]: provides

File: _posts/2017-03-11-Error-Levels.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+but it isn't something to act upon. The application continues even if it's full behaviour isn't quite as expected.
+
+Personally, I wouldn't use this level of logging, as if it's not actionable, people won't pay attention to it. Why not throw an error 
+where the problem occurs, or if possible, guard against the possibility of their being an issue. 
+
+It is used however. An example given in [2](http://stackoverflow.com/questions/2031163/when-to-use-the-different-log-levels) is for logging bad login attempts.
```
Comment (neonboxx) [inline _posts/2017-03-11-Error-Levels.md]: probably reads better as 'in [reference 2]' or 'in [this stackoverflow question (2)]'

File: _posts/2017-03-11-Error-Levels.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+can be useful - but does using info add too much noise to those precious logs we rely on?
+
+The details that could be logged at this level, I tend to log at Debug level. For example, I like to have the elapsed time of inputs and outputs of a service available in logs. 
+Why I chose to log these details at a Debug level instead of Information will be discussed in the next section. 
+
+#### DEBUG
+
+The Debug level is there to help diagnose issues. Again, some developers choose not to use this level of log at all, or filter it out, choosing to avoid the noise.
+In my experience, the debug level has been useful for logging certain information about the application. As mentioned in the information level section, I have seen 
+it used (and used it myself) for logging information such as the elapsed time of a request or time of storing something to a database. The reason I keep this is the logs is
```
Comment (neonboxx) [inline _posts/2017-03-11-Error-Levels.md]: in the logs?

**Comment by neonboxx** [comment]
Looks good, some weird grammar but i'm not smart enough to correct most of it lol. There's some weird line breaks in there too, guessing that's just to make it look right on the page?

File: _posts/2017-03-11-Error-Levels.md (hunk 4 above)
Comment (wordshaker) [inline _posts/2017-03-11-Error-Levels.md]: Haha I don't speak English it turns out

**Comment by wordshaker** [comment]
I've fixed the errors john pointed out. the weird line breaks are now gone as well - it was to make it all fit on my screen when I was typing 😄

**Pull request closed without merging**